### PR TITLE
Speckle tracking

### PR DESCRIPTION
**Pull request by tristan-deep**
Added two point tracking methods that can be used for 2D and 3D B-mode speckle tracking.

```python
from zea.tracking import LucasKanadeTracker, SegmentationTracker
```

Added one tensor ops func:
```python
from zea.tensor_ops import find_contour
```

For example see the added [example notebook](https://zea--170.org.readthedocs.build/en/170/notebooks/models/speckle_tracking_example.html).

tracking_comparison

## Summary by CodeRabbit

* **New Features**
  * Added a tracking module with Lucas–Kanade optical-flow and segmentation-based trackers for 2D/3D motion
  * Added contour detection for binary masks
  * Added an inference-size configuration for segmentation

* **Tests**
  * Added comprehensive tests for tracking, contour detection (including empty-mask behavior) and many edge cases
  * Updated per-notebook test parameters for a sample example notebook

* **Documentation**
  * Added a tutorial reference for the diffusion model

### CODE REVIEW

**Comment by tristan-deep** [comment]
@coderabbitai full review

File: zea/models/lv_segmentation.py [hunk 1]
```diff
@@ -44,6 +44,8 @@
 from zea.models.preset_utils import get_preset_loader, register_presets
 from zea.models.presets import augmented_camus_seg_presets
 
+INFERENCE_SIZE = 256
+
```
[the remark this answers is not in the document]
Comment (tristan-deep) [reply zea/models/lv_segmentation.py]: Better to have the docstring mention explicitely the size (which is fixed). Also the repr has been fixed now.
